index.js

```diff
@@ -0,0 +1,280 @@
+      "region": "Europe",
+      "capital": "Riga"
+  },
+  {
+      "name": "Lebanon",
+      "flag": "https://flagcdn.com/w320/lb.png",
+      "population": 5988000,
+      "region": "Asia",
+      "capital": "Beirut"
+  },
+  {
+      "name": "Lesotho",
+      "flag": "https://flagcdn.com/w320/ls.png",
+      "population": 1894194,
+      "region": "Africa",
+      "capital": "Maseru"
+  },
+  {
+      "name": "Liberia",
+      "flag": "https://flagcdn.com/w320/lr.png",
+      "population": 4615000,
+      "region": "Africa",
+      "capital": "Monrovia"
+  },
+  {
+      "name": "Libya",
+      "flag": "https://flagcdn.com/w320/ly.png",
+      "population": 6385000,
+      "region": "Africa",
+      "capital": "Tripoli"
+  },
+  {
+      "name": "Liechtenstein",
+      "flag": "https://flagcdn.com/w320/li.png",
+      "population": 37623,
+      "region": "Europe",
+      "capital": "Vaduz"
+  },
+  {
+      "name": "Lithuania",
+      "flag": "https://flagcdn.com/w320/lt.png",
+      "population": 2872294,
+      "region": "Europe",
+      "capital": "Vilnius"
+  },
+  {
+      "name": "Luxembourg",
+      "flag": "https://flagcdn.com/w320/lu.png",
+      "population": 576200,
+      "region": "Europe",
+      "capital": "Luxembourg"
+  }
+]
+// console.log(CountriesData);
+// console.log("hi");
+// import countries from './CountriesData.json' 
+//console.log(CountriesData)
+let countriesGrid = document.querySelector(".countries-grid");
+const searchInput=document.querySelector("[data-search]")
+
+searchInput.addEventListener("input",e=>{
+  const value=e.target.value.toLowerCase()
+  //console.log(value)
+  const searchedCountries = CountriesData.filter(
+      (country) => country.name.toLowerCase().includes(value)
+    );
+  //console.log(searchedCountries);
+  countriesGrid.innerHTML = renderCountries(searchedCountries);
+  
+
+})
+
+const renderCountries = (countries) => {
+  return countries.map(
+      (country) =>
+        `
+      <a href="#" class="country scale-effect" data-country-name="${country.name}">
+        <div class="country-flag">
+          <img
+            src="${country.flag}"
+            alt="${country.name} FLag"
+          />
+        </div>
+        <div class="country-info">
+          <h2 class="country-title">${country.name}</h2>
+          <ul class="country-brief">
+            <li>
+              <strong>population: </strong>${country.population}
+            </li>
+            <li>
+              <strong>Region: </strong>${country.region}
+            </li>
+            <li>
+              <strong>capital: </strong>${country.capital}
+            </li>
+          </ul>
+        </div>
+      </a>`
+    )
+    .join("");
+};
+
+const toggleFilterRegionMenu = () => {
+  let filter = document.getElementById("dropdown-body").style;
+
+  filter.visibility === "hidden"
+    ? ((filter.visibility = "visible"), (filter.opacity = 1))
+    : ((filter.visibility = "hidden"), (filter.opacity = 0));
```

Comment:
You can extract the repeating code into a function :


```js 
function toggleVisibility(filter, isVisible) {
  filter.visibility = isVisible ? "visible" : "hidden";
  filter.opacity = isVisible ? 1 : 0;
}


const isCurrentlyHidden = filter.visibility === "hidden";
toggleVisibility(filter, isCurrentlyHidden);

```

But generally use css classes to style elements and not from js